src/main/java/com/aws/greengrass/lifecyclemanager/KernelAlternatives.java

```diff
@@ -105,7 +107,15 @@ public Path getLoaderPath() {
     }
 
     private Path getLoaderPathFromLaunchDir(Path path) {
-        return path.resolve(KERNEL_DISTRIBUTION_DIR).resolve(KERNEL_BIN_DIR).resolve(LOADER_FILE);
+        Path launchPath = path.resolve(KERNEL_DISTRIBUTION_DIR).resolve(KERNEL_BIN_DIR).resolve(LOADER_FILE);
+        if (Files.exists(launchPath, NOFOLLOW_LINKS) && !launchPath.toFile().canExecute()) {
+            try {
+                Platform.getInstance().setPermissions(OWNER_RWX_ONLY, launchPath);
```

Comment:
OWNER_RWX_EVERYONE_RX?

Reply:
And can we only change in `setupInitLaunchDirIfAbsent`? so it's more explicit and only done once?

Reply:
What if the directory is present, but isn't setup correctly? I put it here so that it is guaranteed to update the permissions to be correct upon launch.

Reply:
Updated to put it in `validateLaunchDirSetup` instead.